ngraph/core/src/op/tensor_iterator.cpp

```diff
@@ -37,13 +37,7 @@ bool op::v0::TensorIterator::visit_attributes(AttributeVisitor& visitor)
     visitor.on_attribute("input_descriptions", m_input_descriptions);
     visitor.on_attribute("output_descriptions", m_output_descriptions);
 
-    for (const auto& output_description : m_output_descriptions)
-    {
-        if (auto concat = as_type_ptr<ConcatOutputDescription>(output_description))
-        {
-            m_num_iterations = ((std::abs(concat->m_end - concat->m_start)) / concat->m_part_size);
-        }
-    }
+    set_num_iteratrions_if_no_slice_inputs(); // TODO can be move to validate_and_infer_types
```

Comment:
why adding TODO instead moving it tovalidate_and_infer_types?

Reply:
Because I want to discus this question in this review.